live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequesterTest.java

```diff
@@ -0,0 +1,211 @@
+    private static final String STORAGE_LOCATION = "http://dummy-url";
+    private static final String RESPONSE_BODY_PATTERN = "{\"%s\": \"%s\"}";
+    private static final String RESPONSE_BODY = createResponseBodyWith(STORAGE_LOCATION_PROPERTY, STORAGE_LOCATION);
+
+    CamelContext camelContext;
+    MockEndpoint mockEndpoint;
+
+    PreSignedUrlRequester preSignedUrlRequester;
+
+    @BeforeAll
+    @SneakyThrows
+    void beforeAll()
+    {
+        camelContext = new DefaultCamelContext();
+        preSignedUrlRequester = new PreSignedUrlRequester(camelContext, MOCK_ENDPOINT);
+        camelContext.addRoutes(preSignedUrlRequester);
+        camelContext.start();
+
+        mockEndpoint = camelContext.getEndpoint(MOCK_ENDPOINT, MockEndpoint.class);
+    }
+
+    @AfterAll
+    void afterAll()
+    {
+        camelContext.stop();
+    }
+
+    @Test
+    void testRequestStorageLocation() throws InterruptedException
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        mockEndpointWillRespondWith(STATUS_CODE_201, RESPONSE_BODY);
+        mockEndpointWillExpectInRequestBody(NODE_ID_PROPERTY, NODE_REF, CONTENT_TYPE_PROPERTY, CONTENT_TYPE);
+
+        // when
+        URL url = preSignedUrlRequester.requestStorageLocation(request);
+
+        // then
+        mockEndpoint.assertIsSatisfied();
+        assertThat(url).asString().isEqualTo(STORAGE_LOCATION);
+    }
+
+    @Test
+    void testRequestStorageLocation_invalidResponseStatusCode()
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        mockEndpointWillRespondWith(STATUS_CODE_500);
+
+        // when
+        Throwable thrown = catchThrowable(() -> preSignedUrlRequester.requestStorageLocation(request));
+
+        // then
+        assertThat(thrown)
+                .cause()
+                .isInstanceOf(LiveIngesterRuntimeException.class)
+                .hasMessageContaining("received:", STATUS_CODE_500);
+    }
+
+    @Test
+    void testRequestStorageLocation_missingStorageLocationPropertyInResponse()
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        String responseBodyWithoutUrl = createResponseBodyWith("unexpectedProperty", STORAGE_LOCATION);
+        mockEndpointWillRespondWith(STATUS_CODE_201, responseBodyWithoutUrl);
+
+        // when
+        Throwable thrown = catchThrowable(() -> preSignedUrlRequester.requestStorageLocation(request));
+
+        // then
+        assertThat(thrown)
+                .cause()
+                .isInstanceOf(LiveIngesterRuntimeException.class)
+                .hasMessageContaining("Missing", STORAGE_LOCATION_PROPERTY);
+    }
+
+    @Test
+    void testRequestStorageLocation_invalidJsonResponse()
+    {
+        // given
+        StorageLocationRequest request = createStorageLocationRequestMock();
+        String invalidJsonBody = RESPONSE_BODY.replaceAll(".$", "");
```

Comment:
Is this the same as removing the last character `}`? If so then it seems like a complex way to write it!

Reply:
Yes, exactly - I've rewrite it making it a bit neater.